python/cuml/manifold/umap.pyx

```diff
@@ -410,6 +407,11 @@ class UMAP(Base):
         def curve(x, a, b):
             return 1.0 / (1.0 + a * x ** (2 * b))
 
+        if has_scipy():
+            from scipy.optimize import curve_fit
+        else:
+            raise RuntimeError('Scipy is needed to run find_ab_params')
```

Comment:
Maybe we should raise ImportErrors for these? RuntimeError is very general... but I'm open if you feel it's best as long as we are consistent

Reply:
Actually, ignore that... I see elsewhere that we use runtimeerror so you're being perfectly consistent.